dspace-spring-rest/src/main/java/org/dspace/app/rest/submit/step/validation/MetadataValidation.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.submit.step.validation;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.dspace.app.rest.model.ErrorRest;
+import org.dspace.app.rest.repository.WorkspaceItemRestRepository;
+import org.dspace.app.rest.submit.SubmissionService;
+import org.dspace.app.util.DCInput;
+import org.dspace.app.util.DCInputSet;
+import org.dspace.app.util.DCInputsReader;
+import org.dspace.app.util.DCInputsReaderException;
+import org.dspace.app.util.SubmissionStepConfig;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.WorkspaceItem;
+import org.dspace.content.authority.service.MetadataAuthorityService;
+import org.dspace.content.service.ItemService;
+
+/**
+ * 
+ * Execute three validation check on fields validation:
+ * - mandatory metadata missing
+ * - regex missing match
+ * - authority required metadata missing
+ * 
+ * @author Luigi Andrea Pascarelli (luigiandrea.pascarelli at 4science.it)
+ *
+ */
+public class MetadataValidation extends AbstractValidation {
```

Comment:
Would it be possible to move this class to `dspace-api`? It looks like "metadata validation" is business logic and is not directly related to the REST API.

Reply:
At moment this class contains logic to transport error message to the client. Usually metadata validation in DSpace is a prerogative of the UI and the batch for example assume that the inserted values are pre validated.